docs/api/ref/GetAgent.md

```diff
@@ -0,0 +1,253 @@
+ * Desktop Agent is detected. There may be additional set-up steps to perform 
+ * which will happen outside the timeout. 
+ * 
+ * @property {boolean} channelSelector Flag indicating that the application  
+ * needs access to a channel selector UI (i.e. because it supports User Channels 
+ * and does not implement its own UI for selecting channels). If not set will 
+ * default to true. MAY be ignored by Desktop Agent Preload (container)  
+ * implementations. 
+ * 
+ * @property {boolean} intentResolver Flag indicating that the application  
+ * needs access to an intent resolver UI (i.e. because it supports raising on or 
+ * more intents and and does not implement its own UI for selecting target apps. 
+ * If not set will default to true. MAY be ignored by Desktop Agent Preload  
+ * (container) implementations. 
+ * 
+ * @property {boolean} dontSetWindowFdc3 For backwards compatibility, `getAgent` 
+ * will set a reference to the Desktop Agent implementation at `window.fdc3` 
+ * if one does not already exist, and will fire the fdc3Ready event. Setting  
+ * this flag to `false` will inhibit that behavior, leaving `window.fdc3` unset. 
+ *  
+ * @property {function} failover An optional function that provides a  
+ * means of connecting to or starting a Desktop Agent, which will be called 
+ * if no Desktop Agent is detected. Must return either a Desktop Agent  
+ * implementation directly (e.g. by using a proprietary adaptor) or a  
+ * WindowProxy (e.g a reference to another window returned by `window.open`  
+ * or an iframe's `contentWindow`) for a window or frame in which it has loaded 
+ * a Desktop Agent or suitable proxy to one that works with FDC3 Web Connection 
+ * and Desktop Agent Communication Protocols. 
+ */ 
+type GetAgentParams = { 
+    timeout?: number, 
+    identityUrl?: string, 
+    channelSelector?: boolean, 
+    intentResolver?: boolean,
+    dontSetWindowFdc3?: boolean,
+    failover?: (args: GetAgentParams) => Promise<WindowProxy | DesktopAgent> 
+};
+```
+
+:::note
+
+As web applications can navigate to or be navigated by users to different URLs and become different applications, validation of apps identity is often necessary. The web application's current URL is passed to web browser-based Desktop Agents to allow them to establish the app's identity - usually connecting it with an App Directory record already known to the Desktop Agent. For more details on identity validation see the identity validation section of the  [Web Connection Protocol (WCP)](specs/webConnectionProtocol).
+
+:::
+
+Finally, if there is still no Desktop Agent available, or an issue prevent connection to it, the `getAgent()` function will reject its promise with a message from the `AgentError` enumeration.
+
+```ts
+/** 
+ * Contains constants representing the errors that can be encountered when  
+ * trying to connect to a web-based Desktop Agent with the getAgent function. 
+ */ 
+enum AgentError { 
```

Comment:
I think it was a mistake in FDC3 to standardise the errors as this limits the expressivity of the error messages coming back to the user.  

I can see this maybe is consistent with what's come before but I'm not sure it's a great idea.

Reply:
Some level of standardisation is needed for intelligent handling of errors (at least this way you can differentiate predefined errors) - but yes I don't lack the lack of additional detail. However, if the original authors of FDC3 had used the javascript Error object properly they could have used the `name` field, leaving `message` open for more details. That would have meant creating an Error type for each possible error. 

As it stands (with error type names used as the `message` , a Desktop Agent can only provide more details through its own logs, rather than loggin in the client window's console, surfacing an error object with more details.

Not an issue with this PR but a general comment (right @robmoffat?) - it is consistent in style with the rest of FDC3.